core/unix/tls_linux_riscv64.c

```diff
@@ -48,29 +48,46 @@
 byte **
 get_dr_tls_base_addr(void)
 {
-    /* FIXME i#3544: Not implemented */
-    ASSERT_NOT_IMPLEMENTED(false);
-    return NULL;
+    byte *lib_tls_base = (byte *)read_thread_register(TLS_REG_LIB);
+    if (lib_tls_base == NULL)
+        return NULL;
+    return (byte **)(lib_tls_base + DR_TLS_BASE_OFFSET);
 }
 
 void
 tls_thread_init(os_local_state_t *os_tls, byte *segment)
 {
-    /* FIXME i#3544: Not implemented */
-    ASSERT_NOT_IMPLEMENTED(false);
+    ASSERT(read_thread_register(TLS_REG_LIB) != 0);
```

Comment:
s/0/NULL/

Reply:
`read_thread_register` returns a `ptr_uint_t`, which is a `uint64`, so we can't use `NULL` here.